pkg/resource/clickpipe_cdc_infrastructure.go

```diff
@@ -0,0 +1,303 @@
+		Attributes: map[string]schema.Attribute{
+			"service_id": schema.StringAttribute{
+				MarkdownDescription: "ClickHouse Cloud service ID where the CDC infrastructure is located.",
+				Required:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"replica_cpu_millicores": schema.Int64Attribute{
+				MarkdownDescription: "CPU in millicores for DB ClickPipes. Must be a multiple of 1000, between 1000 and 24000.",
+				Required:            true,
+				Validators: []validator.Int64{
+					int64validator.Between(1000, 24000),
+				},
+			},
+			"replica_memory_gb": schema.Float64Attribute{
+				MarkdownDescription: "Memory in GiB for DB ClickPipes. Must be a multiple of 4, between 4 and 96. Must be 4× the CPU core count.",
+				Required:            true,
+				Validators: []validator.Float64{
+					float64validator.Between(4, 96),
+				},
+			},
+		},
+	}
+}
+
+func (r *ClickPipeCdcInfrastructureResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	if req.ProviderData != nil {
+		client, ok := req.ProviderData.(api.Client)
+		if !ok {
+			resp.Diagnostics.AddError(
+				"Unexpected Resource Configure Type",
+				fmt.Sprintf("Expected api.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+			)
+			return
+		}
+		r.client = client
+	}
+}
+
+func (r *ClickPipeCdcInfrastructureResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan models.ClickPipeCdcInfrastructureModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serviceID := plan.ServiceID.ValueString()
+
+	// Validate CPU is a multiple of 1000
+	cpuMillicores := plan.ReplicaCpuMillicores.ValueInt64()
+	if cpuMillicores%1000 != 0 {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_cpu_millicores must be a multiple of 1000, got %d", cpuMillicores),
+		)
+		return
+	}
+
+	// Validate memory is a multiple of 4
+	memoryGb := plan.ReplicaMemoryGb.ValueFloat64()
+	if int(memoryGb)%4 != 0 {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_memory_gb must be a multiple of 4, got %.1f", memoryGb),
+		)
+		return
+	}
+
+	// Validate memory is 4x CPU cores
+	cpuCores := float64(cpuMillicores) / 1000.0
+	expectedMemory := cpuCores * 4
+	if memoryGb != expectedMemory {
+		resp.Diagnostics.AddError(
+			"Invalid Configuration",
+			fmt.Sprintf("replica_memory_gb must be 4× the CPU core count. With %d millicores (%.1f cores), memory must be %.1f GB, but got %.1f GB",
+				cpuMillicores, cpuCores, expectedMemory, memoryGb),
+		)
+		return
+	}
+
+	// Poll for CDC scaling endpoint to become available (up to 10 minutes)
+	maxWait := 10 * time.Minute
+	checkAvailable := func() error {
+		_, err := r.client.GetClickPipeCdcScaling(ctx, serviceID)
```

Comment:
why do we need this at all rather than failing the apply when users tries to create a cdc_infrastructure but has no pipes?